frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -90,6 +91,68 @@ export const experimentLogic = kea<experimentLogicType>({
             },
         ],
     }),
+    selectors: ({ values }) => ({
+        breadcrumbs: [
+            (s) => [s.experimentData, s.experimentId],
+            (experimentData, experimentId): Breadcrumb[] => [
+                {
+                    name: 'Experiments',
+                    path: urls.experiments(),
+                },
+                {
+                    name: experimentData?.name || 'New Experiment',
+                    path: urls.experiment(experimentId || 'new'),
+                },
+            ],
+        ],
+        funnel: [
+            (s) => [
+                funnelLogic({ dashboardItemId: values.experimentFunnelId, syncWithUrl: false }).selectors.results,
+                s.newExperimentData,
+            ],
+            (results) => {
+                // eslint-disable-line
+                const newResults = funnelLogic.findMounted({ dashboardItemId: values.experimentFunnelId })?.values
+                    .results // valid results
+                const newResultsWithoutFound = funnelLogic({
```

Comment:
You shouldn't use `logic.values` in the selector array directly, because that's only evaluated once at logic init, so in this case `dashboardItemId: values.experimentFunnelId` probably is null. Instead, you can make use of the fact that selectors are just functions that spit out the value based on a snapshot of the global state + local props. So instead of
```JS
funnelLogic({ dashboardItemId: values.experimentFunnelId, syncWithUrl: false }).selectors.results
```
try
```JS
(state, props) => funnelLogic.selectors.results(
	state,
	{ dashboardItemId: s.experimentFunnelId(state, props), syncWithUrl: false }
)
```

Reply:
That makes a lot of sense, and explains why the results earlier are invalid, thanks! :)

I tried this and:

![image](https://user-images.githubusercontent.com/7115141/146009936-3bbefaf6-1673-484d-84df-a694f378f32f.png)

It seems like `funnelLogic.selectors` is not defined 🤔 .  And if I do: `funnelLogic.selectors?.results` I run into the same problem: evaluated once, stays undefined forever

Reply:
This isn't yet a specific keyed logic, so how do I ensure its mounted and available? 👀

Reply:
Hmmm, so logic `connect` can't be used here because that can only be parametrized with props. And `.mount()` in a selector would be really dirty + wouldn't be able to unmount. What if you try mounting `funnelLogic` at the component level with `useMountedLogic(funnelLogic({ dashboardItemId: experimentFunnelId, syncWithUrl: false }))`?

Reply:
That's interesting! Didn't realise this existed.

So, here: https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/experiments/Experiment.tsx#L48

I'm using `useValues` for this `funnelLogic`. Shouldn't that imply I don't need to use `useMountedLogic` here?

Reply:
Feels like this is turning out to be more trouble than its worth. I would've thought I'm doing something simple wrong :$

I can just let the selectors return functions and use those with the values from this mounted component^^

Reply:
Yeah, that might just be the optimal thing 😄